test/jdk/tools/jpackage/helpers/jdk/jpackage/test/TKit.java

```diff
@@ -233,6 +233,43 @@ public static void createTextFile(Path filename, Stream<String> lines) {
         trace("Done");
     }
 
+    public static void createJPackageXMLFile(Path appImageDir,
+            String mainLauncher, String mainClass) throws IOException {
+        String version = System.getProperty("java.version");
+        String platform = "";
+        Path appDir = appImageDir;
+        if (TKit.isWindows()) {
+            platform = "windows";
+            appDir = Files.createDirectories(
+                         appImageDir.resolve("app"));
+        } else if (TKit.isLinux()) {
+            platform = "linux";
+            appDir = Files.createDirectories(
+                         appImageDir.resolve("lib").resolve("app"));
+        } else if (TKit.isOSX()) {
+            platform = "macOS";
+            appDir = Files.createDirectories(
+                         appImageDir.resolve("Contents").resolve("app"));
+        }
+
+        TKit.createTextFile(appDir.resolve(".jpackage.xml"), List.of(
+                            "<?xml version=\"1.0\" encoding=\"utf-8\"?>",
+                            String.format(
+                                "<jpackage-state platform=\"%s\" version=\"%s\">",
+                                platform, version),
+                            "<app-version>1.0</app-version>",
+                            String.format(
+                                "<main-launcher>%s</main-launcher>",
+                                mainLauncher),
+                            String.format(
+                                "<main-class>%s</main-class>",
+                                mainClass),
+                            "<signed>false</signed>",
+                            "<app-store>false</app-store>",
+                            "</jpackage-state>"
+            ));
+    }
+
```

Comment:
1. This is jpackage specific, so it should belong to JPackageCommand class.
2. Please use proper XML API to create XML output.
3. Having `<app-version>1.0</app-version>` in .jpackage.xml file will make `AppImageFile.isValid()` return `false` and `AppImageFile.load()` to throw exception. Which seems to be equivalent of not having .jpackage.xml file in app image at all making it a question if this new function is need at all.

Reply:
1 and 2 done. I think you mistaken `<app-version>1.0</app-version>` with `<jpackage-state version=\"%s\"`. We never read <app-version> and I do put correct value for `<jpackage-state version=\"%s\"`. It is required for several tests which uses empty/dummy app images and since we started required .jpackage.xml file we need it for empty app images as well.

Reply:
Right! My bad, sorry for the confusion.